system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -351,39 +352,82 @@ class CLdapConfig : implements ILdapConfig, public CInterface
 
         m_timeout = cfg->getPropInt(".//@ldapTimeoutSecs", LDAPTIMEOUT);
 
+        //------------------------------------------------
+        //Get LDAP Admin account username and password
+        // m_sysuser_commonname
+        // m_sysuser_password
+        //------------------------------------------------
+//@@
+        StringBuffer xml;//@@
+        toXML(cfg, xml);//@@
+        DBGLOG("@@%s",xml.str());//@@
+//@@
+        StringBuffer systemUserSecretKey;
+        cfg->getProp(".//@systemUserSecretKey", systemUserSecretKey);//secrets repo key
+        if (!systemUserSecretKey.isEmpty())
+        {
+            StringBuffer sb;
+            cfg->getProp(".//@systemCommonName", sb);
+            if (!sb.isEmpty())
+                WARNLOG("Please specify either systemCommonName or systemUserSecretKey in configuration");
+
+            StringBuffer vaultId;
+            cfg->getProp(".//@systemUserVaultID", vaultId);//optional HashiCorp vault ID
+
+            DBGLOG("Retrieving LDAP Admin username/password from secrets repo: %s %s", !vaultId.isEmpty() ? vaultId.str() : "", systemUserSecretKey.str());
```

Comment:
StringBuffer::str() won't return nullptr so 
```
!vaultId.isEmpty() ? vaultId.str() : ""
```
is redundant, just "vaultId.str()" should be enough.